### PR TITLE
media-libs/allegro: update to 4.4.3.1

### PR DESCRIPTION
**Pull request by winterheart**
Updated to new maintainance release. Some patches applied by upstream.
Added multilib support for 4.2 slot. Migrated to EAPI=7 and cmake eclass.
    
Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: Azamat H. Hackimov <azamat.hackimov@gmail.com>

### CODE REVIEW

File: media-libs/allegro/allegro-4.4.2-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	X? (
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXcursor[${MULTILIB_USEDEP}]
+		x11-libs/libXext[${MULTILIB_USEDEP}]
+		x11-libs/libXpm[${MULTILIB_USEDEP}]
+		x11-libs/libXt[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86dga[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86vm[${MULTILIB_USEDEP}]
+		opengl? (
+			virtual/glu
```
Comment (a17r) [inline media-libs/allegro/allegro-4.4.2-r3.ebuild]: missing `[${MULTILIB_USEDEP}]`?

File: media-libs/allegro/allegro-4.4.2-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			virtual/opengl
```
Comment (a17r) [inline media-libs/allegro/allegro-4.4.2-r3.ebuild]: missing `[${MULTILIB_USEDEP}]`?

File: media-libs/allegro/allegro-4.4.2-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (a17r) [inline media-libs/allegro/allegro-4.4.2-r3.ebuild]: What about EAPI-7?

File: media-libs/allegro/allegro-4.4.2-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		-DWANT_JPGALLEG=$(usex jpeg)
+		-DWANT_LINUX_CONSOLE=OFF
+		-DWANT_LINUX_FBCON=$(usex fbcon)
+		-DWANT_LINUX_SVGALIB=$(usex svga)
+		-DWANT_LINUX_VGA=$(usex vga)
+		-DWANT_LOADPNG=$(usex png)
+		-DWANT_LOGG=$(usex vorbis)
+		-DWANT_OSS=$(usex oss)
+		-DWANT_TESTS=$(usex test)
+		-DWANT_TOOLS=$(usex X)
```
Comment (a17r) [inline media-libs/allegro/allegro-4.4.2-r3.ebuild]: should this maybe be native-abi only, if it installs binaries?

File: media-libs/allegro/allegro-4.4.2-r3.ebuild [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+SRC_URI="mirror://sourceforge/alleg/${P}.tar.gz"
+
+LICENSE="Allegro MIT GPL-2+ ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~mips ~ppc ~ppc64 ~x86"
+IUSE="alsa fbcon jack jpeg opengl oss png svga test vga vorbis X"
+
+RDEPEND="
+	alsa? ( media-libs/alsa-lib[${MULTILIB_USEDEP}] )
+	jack? ( media-sound/jack-audio-connection-kit[${MULTILIB_USEDEP}] )
```
Comment (a17r) [inline media-libs/allegro/allegro-4.4.2-r3.ebuild]: Could it depend on `virtual/jack` instead?

File: media-libs/allegro/allegro-4.4.2-r3.ebuild (hunk 1 above)
Comment (winterheart) [reply media-libs/allegro/allegro-4.4.2-r3.ebuild]: Done.

File: media-libs/allegro/allegro-4.4.2-r3.ebuild (hunk 2 above)
Comment (winterheart) [reply media-libs/allegro/allegro-4.4.2-r3.ebuild]: Done.

File: media-libs/allegro/allegro-4.4.2-r3.ebuild (hunk 3 above)
Comment (winterheart) [reply media-libs/allegro/allegro-4.4.2-r3.ebuild]: Done.

File: media-libs/allegro/allegro-4.4.2-r3.ebuild (hunk 4 above)
Comment (winterheart) [reply media-libs/allegro/allegro-4.4.2-r3.ebuild]: This part already on multilib_is_native_abi part on installation step.

File: media-libs/allegro/allegro-4.4.2-r3.ebuild (hunk 5 above)
Comment (winterheart) [reply media-libs/allegro/allegro-4.4.2-r3.ebuild]: Done.

File: media-libs/allegro/allegro-4.4.3.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+		x11-libs/libXt[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86dga[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86vm[${MULTILIB_USEDEP}]
+		opengl? (
+			virtual/glu[${MULTILIB_USEDEP}]
+			virtual/opengl[${MULTILIB_USEDEP}]
+		)
+	)"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (chewi) [inline media-libs/allegro/allegro-4.4.3.1.ebuild]: pkgconfig goes in BDEPEND under EAPI 7.

Comment (winterheart) [reply media-libs/allegro/allegro-4.4.3.1.ebuild]: @chewi done

**Comment by juippis** [comment]
Is this PR still relevant after cb16966c9885f4a3f8a9f257dc890853bd1e46de

**Comment by winterheart** [comment]
@juippis I don't see any relevance with these changes. Moreover, I suspect these fixes already fixed in Allegro 4.4.3.x, which is not in tree yet and included to this PR.

**Comment by juippis** [comment]
Right, there's a version bump. You introduce *allegro-4.4.2-GLX_RGBA_FLOAT_BIT.patch* which is identical to what got pushed and it caught my eye. So you can just use the previous patch without having to upload a new identical one.

**Comment by winterheart** [comment]
Ok, some squashes and updates done.

**Comment by winterheart** [comment]
Removed test USE-flag since it does nothing regarding unittests.

**Comment by chewi** [comment]
I thought I better not let this reach its 1st birthday but the build failed for me:

```
FAILED: docs/info/allegro.info 
cd /var/tmp/portage/media-libs/allegro-4.4.3.1/work/allegro-4.4.3.1_build-abi_x86_32.x86/docs && /usr/bin/makeinfo --no-split -o /var/tmp/portage/media-libs/allegro-4.4.3.1/work/allegro-4.4.3.1_build-abi_x86_32.x86/docs/info/allegro.info /var/tmp/portage/media-libs/allegro-4.4.3.1/work/allegro-4.4.3.1_build-abi_x86_32.x86/docs/texi/allegro.texi
utf8 "\xB0" does not map to Unicode at /usr/share/texinfo/Texinfo/ParserNonXS.pm line 1796, <FH> line 18520.
Malformed UTF-8 character: \xb0 (unexpected continuation byte 0xb0, with no preceding start byte) in pattern match (m//) at /usr/share/texinfo/Texinfo/ParserNonXS.pm line 3364.
Malformed UTF-8 character (fatal) at /usr/share/texinfo/Texinfo/ParserNonXS.pm line 3364.
```

This looks vaguely familiar but need sleep before I can investigate. Help appreciated.

**Comment by chewi** [comment]
Done.

I knew I'd seen the above issue before. I'd just disabled the docs last time but I found a patch from Red Hat this time. I fixed the automagic building of these docs while I was at it.

I tried some revdeps. `games-action/chickens` failed to build but turns out it was already broken. The rest that I tried did work.

**Pull request closed without merging**